app/views/pages/use-cases/using-rest-endpoints-with-react.liquid

```diff
@@ -0,0 +1,300 @@
+
+### Step 4: Prepare your endpoints' headers
+
+As you will receive json with your endpoint, you need to create a liquid file under `app/views/partials/shared/api/headers.liquid` to render the necessary headers:
+
+```liquid
+{
+  "Content-Type": "application/json"
+}
+```
+
+### Step 5: Create your model's schema
+
+You can define your data models under `app/model_schemas/`. In this example you will create basic users so create `app/model_schemas/user.yml`:
+
+```yml
+name: user
+properties:
+  - name: name
+    type: string
+  - name: email
+    type: string
+```
+
+### Step 6: Create a REST endpoint to store a new user
+
+You need to create a graphql mutations to store your data.You can collect these graphql files under `app/graphql/` folder.
+
+As you want to store a new user you need to create your mutation under `app/graphql/users/create_user.graphql`:
+
+```graphql
+mutation create_user($name: String!, $email: String!) {
+  user: model_create(
+    model: {
+      model_schema_name: "user"
+      properties: [
+        { name: "name", value: $name }
+        { name: "email", value: $email }
+      ]
+    }
+  ) {
+    id
+  }
+}
+```
+
+After that you can define a new page in your platformOS instance and use that as a REST endpoint of your React app. You can define new pages by creating new liquid files under `app/views/pages/` but it's recommended to collect your REST files in an `api` subfolder.
+
+So you can define a new REST endpoint to store your users by creating `app/views/pages/api/users/post.liquid`:
+
+```liquid
+---
+slug: api/users
+method: post
+layout: ''
+response_headers: >
+  {% render 'shared/api/headers' %}
+---
+
+{% liquid
+  graphql result = 'users/create_user', name: context.params.name, email: context.params.email
+  unless result.user
+    response_status 400
+  endunless
+%}
+
+{{ result }}
+```
+
+As you can see, you can define your endpoint's slug and method, you don't need a layout and you can use your previously created `headers.liquid` file to render you response header.
+
+In the logic part, you can use your previously created graphql file with the parameters from the context and based on that's result you can set an error status code. As the result is in `json` format, just render it and it will contain everything.
+
+### Step 7: Use your REST endpoint in your React app
+
+Now you have everything to store a user so you can modify your React app's `/my-app/src/App.tsx` and append a form on it. In this example you will use typescript and react hook form to implement the POST request and you can see the created user's ID in the JavaScript console.
+
+We _strongly recommend_ to use [axios](https://github.com/axios/axios) as the `csrf-token` is very important and it's easy to set up with:
+
+```javascript
+axios.defaults.xsrfCookieName = 'CSRF-TOKEN';
+axios.defaults.xsrfHeaderName = 'X-CSRF-Token';
+```
+
+So install and use that instead of `fetch`
+
+```shell
+/my-app$ npm install axios --save
+```
```

Comment:
Can’t agree with this whole section, but I won’t force it. `fetch` is great and standard.

Reply:
Let's discuss it with @streflik